crates/polars-core/src/chunked_array/from.rs

```diff
@@ -206,8 +206,8 @@ where
         // that check if the data types in the arrays are as expected
         #[cfg(debug_assertions)]
         {
-            if !chunks.is_empty() && dtype.is_primitive() {
-                assert_eq!(chunks[0].data_type(), &dtype.to_physical().to_arrow(true))
```

Comment:
The `.to_physical()` here is unneeded as we have already checked that `dtype` is primitive.